lib/ts/PriorityQueue.h

```diff
@@ -123,11 +129,18 @@ PriorityQueue<T, Comp>::erase(PriorityQueueEntry<T> *entry)
     return;
   }
 
-  _v[entry->index] = _v[_v.length() - 1];
-  _v.pop();
-  _bubble_down(entry->index);
-  if (!empty()) {
-    _bubble_up(entry->index);
+  ink_release_assert(entry->index < _v.length());
+  unsigned int original_index = entry->index;
+  if (original_index != (_v.length() - 1)) {
+    // Move the erased item to the end to be popped off
+    _swap(original_index, _v.length() - 1);
+    _v.pop();
+    _bubble_down(original_index);
+    if (!empty()) {
```

Comment:
Agreed the !empty() check doesn't seem to be necessary.  I just copied the block around.  @jacksontj  was there a reason you remember for making the !empty() check?  

I'll remove it and give it a try tomorrow.

The bubble_down makes sure the specified index is not greater than anything higher in the array.  bubble_up makes sure the specified index is not lesser than any of the indices lower in the array.  If you are just adding to adding to one end, you only need to check one side, so only one of the calls needs to be made.  Seems like you could bundle this into a combined call and look at the new index to decide which version to call, but I'd rather not make any more semi-substantive changes here at this point.